src/frequenz/client/common/microgrid/__init__.py

```diff
@@ -1,4 +1,51 @@
 # License: MIT
-# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
 
-"""Frequenz microgrid definition."""
+"""Module to define the microgrid used with the common client."""
+
+from __future__ import annotations  # required for constructor type hinting
+
+from dataclasses import dataclass
+from typing import List, Self
+
+# pylint: disable=import-error, no-name-in-module
+from frequenz.api.common.v1.microgrid.microgrid_pb2 import (
+    MicrogridComponentIDs as PBMicrogridComponentIDs,
+)
+
+
+@dataclass(frozen=True, kw_only=True)
+class MicrogridComponentIDs:
+    """Linking component IDs with their respective microgrid ID."""
+
+    microgrid_id: int
+    """The ID of the microgrid."""
+
+    component_ids: List[int]
+    """List of component IDs belonging to this microgrid."""
+
+    @classmethod
+    def from_proto(cls, microgrid_component_ids: PBMicrogridComponentIDs) -> Self:
+        """Convert a protobuf MicrogridComponentIDs to MicrogridComponentIDs object.
+
+        Args:
+            microgrid_component_ids: MicrogridComponentIDs to convert.
+
+        Returns:
+            MicrogridComponentIDs object corresponding to the protobuf message.
+        """
+        return cls(
+            microgrid_id=microgrid_component_ids.microgrid_id,
+            component_ids=List(microgrid_component_ids.component_ids),
```

Comment:
This raises an interesting question, copying the list (or whatever `component_ids` is in the proto class) is definitely the safe option, but it means doing some copying, when in 99% of the cases you'll probably just take the protobuf message, convert it to the wrapper and discard it, so it is very unlikely it will be mutated after creating the wrapper (in which case it will also affect the wrapper list, as we stored a reference to the protobuf message).

Maybe since it is unlikely and might have a performance impact, we could document that a reference is taken and the users should take care of the copying if they plan to keep mutating the proto message instance.

@frequenz-floss/python-sdk-team any opinions?

Reply:
Haven't really figured out what this type is for, but it still looks like a one time thing that we might have to process at startup, and not along with each streaming message, etc.  So I don't see any benefit from not choosing the conservative approach.

Reply:
Yeah, this is a more general question, because we will probably have it in other situations, as for this particular class, I guess we need to figure out if we want to pursue this PR as you mentioned in https://github.com/frequenz-floss/frequenz-client-common-python/pull/8#issuecomment-1954809138, I think it was a good point that we might be missing the end goal here.

Reply:
I agree it is fine to copy the IDs if the message is only processed once at startup (or only a few times). For general scenarios it makes sense to avoid the copy if it is possible